dali/pipeline/operators/color/hsv.cc

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/operators/color/hsv.h"
+#include "dali/kernels/imgproc/pointwise/linear_transformation_cpu.h"
+
+namespace dali {
+namespace {
+
+template <typename Out, typename In>
+using TheKernel = kernels::LinearTransformationCpu<Out, In, 3, 3, 3>;
+}  // namespace
+
+
+DALI_SCHEMA(Hsv)
+                .DocStr(R"code(This operator performs HSV manipulation.
+To change hue, saturation and/or value of the image, pass corresponding deltas.
+Keep in mind, that `hue` has additive delta argument,
+while for `saturation` and `value` they are multiplicative.
+
+This operator accepts RGB color space as an input.
+)code")
+                .NumInput(1)
+                .NumOutput(1)
+                .AddOptionalArg(hsv::kHue,
+                                R"code(Set additive brightness delta. 0 denotes no-op)code", .0f,
+                                true)
+                .AddOptionalArg(hsv::kSaturation,
+                                R"code(Set multiplicative contrast delta. 1 denotes no-op)code",
```

Comment:
I've never heard of "multiplicative delta". Neither has google.

Reply:
Done